### PR TITLE
Fixing broken Gcm token validation.

### PR DESCRIPTION
**Pull request by centerorbit**
This is to resolve this issue:
https://github.com/Ph3nol/NotificationPusher/issues/64

### CODE REVIEW

**Comment by Ph3nol** [comment]
According to me, OK this could fix the problem, but for me it's not enough restrictive.
Is it possible to allow an exhaustive list of patterns, an adapted regex or other king of things like this?

Don't forget to adapt tests assertions to your changes.

Thank you so much for your PR.

**Comment by centerorbit** [comment]
@Ph3nol I've looked into this a bit more, and have not been able to find a documented pattern. There is also this answer on SO: http://stackoverflow.com/a/12502351/663058

My original idea was to make a regex that matched current and past pattern. Something like this:
`   /^([0-9a-zA-Z\-\_]+:)?[0-9a-zA-Z\-\_]+$/i `
... but without knowing what it could be next, it could break in the future.

So I'm going make another commit to adapt your tests, and then change the line to ensure it's a string, instead of just not null:
` return is_string($token) && $token!= ''; `

**Comment by centerorbit** [comment]
@Ph3nol Made a change to how the token is validated: no numbers, objects, arrays. Only not empty strings. Updated tests to match, ran them. And here we are now! Let me know if you need anything else.

**Comment by thbsolutions** [comment]
Please, merge is as soon as possible, I also have this issue.

File: src/Sly/NotificationPusher/Adapter/Gcm.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ class Gcm extends BaseAdapter
      */
     public function supports($token)
     {
-        return (bool) preg_match('/^[0-9a-zA-Z\-\_]+$/i', $token);
+	    return is_string($token) && $token!= '';
```
Comment (Ph3nol) [inline src/Sly/NotificationPusher/Adapter/Gcm.php]: Hello,
Can you please respect basic PHP coding styles and PSR-2 standards?
No tabs but spaces (1 tab = 4 spaces), spaces before and after operators etc.
http://www.php-fig.org/psr/psr-2/fr/

I'll merge your PR after your corrections.

Thank you very much.

File: tests/units/Sly/NotificationPusher/Adapter/Gcm.php [hunk 2]
```diff
@@ -21,6 +21,9 @@
 class Gcm extends Units\Test
 {
     const GCM_TOKEN_EXAMPLE = 'AAA91bG9ISdL94D55C69NplFlxicy0iFUFTyWh3AAdMfP9npH5r_JQFTo27xpX1jfqGf-aSe6xZAsfWRefjazJpqFt03Isanv-Fi97020EKLye0ApTkHsw_0tJJzgA2Js0NsG1jLWsiJf63YSF8ropAcRp4BSxVBBB';
+    // The format of GCM tokens apparently have changed,
+    //   this string looks similar to new format
```
Comment (Ph3nol) [inline tests/units/Sly/NotificationPusher/Adapter/Gcm.php]: Why 2 lines for this comment?

**Comment by ckopnio** [comment]
Hello,
please, merge is as soon as possible, I also have this issue.
Thank

**Comment by Ph3nol** [comment]
Hi,
I'll merge it after @centerorbit last coding standards update (see my comments about its last commit).
Wait and see! ;)

**Comment by thiagostropp** [comment]
Cmom @centerorbit, people are hardcoding stuff here! :P

File: src/Sly/NotificationPusher/Adapter/Gcm.php (hunk 1 above)
Comment (centerorbit) [inline src/Sly/NotificationPusher/Adapter/Gcm.php]: Fixed

File: tests/units/Sly/NotificationPusher/Adapter/Gcm.php (hunk 2 above)
Comment (centerorbit) [inline tests/units/Sly/NotificationPusher/Adapter/Gcm.php]: Fixed

**Comment by centerorbit** [comment]
I've adjusted as requested.

**Comment by Ph3nol** [comment]
Perfect, thank you so much! :smile: :+1:

**Comment by Ph3nol** [comment]
Hey @centerorbit, @thbsolutions, @ckopnio, @akimapilot,
I've just merged this PR and tagged a `v2.2.10` release.
Enjoy! :smile:

**Comment by sethdaniel** [comment]
We are getting this message.  Went and checked the supports($token) function in GCM.php and sure enough that function is only validating that the token is not empty.  If we are getting this error message and the function is only checking for not empty string is it safe to assume that we are getting the message because the token is, indeed, empty?
